### ISSUE TITLE
Any way to customize facet position?

### ISSUE THREAD

**Issue by thiagoveloso** [thomasp85/patchwork#205]
I am plotting a `ggplot2` graph that has an uneven number of facets. My plot currently looks like this:

```
library(ggplot2)

df <- data.frame(group=c(1,1,2,2,3),
                 name=c('a','b','c','d','e'),
                 x=c(1,2,3,4,5),
                 y=c(2,3,4,5,6))
ggplot(df, aes(x,y)) + geom_point() + facet_wrap(~ name, ncol=3) 
```

[![enter image description here][1]][1]

Note that the default positioning for the facets is:

```
a b c
d e
```

However, I would like to have a finer control on the positioning of the facets. Specifically, I would like to "center" the bottom two facets in order to distribute the void space more evenly on the sides of the plot.

In other words, the ideal placement for me would look like:

```
a b c
 d e
```

Is this something I can achieve with `patchwork`? How?

  [1]: https://i.stack.imgur.com/Qkq50.png

**Comment by thomasp85** [comment]
No - patchwork does not modify individual facets in a plot

**Comment by mattwarkentin** [comment]
I know this thread is long dead, but for anyone who stumbles upon this at a future time - you can definitely create the pattern desired in the original post relatively simply by creating multiple ggplots and then using `patchwork`. I actually use this pattern quite often to achieve the facet feel with flexible layouts via `patchwork`.
``` r
library(dplyr)
library(ggplot2)
library(patchwork)

df <- data.frame(group=c(1,1,2,2,3),
                 name=c('a','b','c','d','e'),
                 x=c(1,2,3,4,5),
                 y=c(2,3,4,5,6))

p1 <-
  df %>% 
  filter(name %in% c("a", "b", "c")) %>% 
  ggplot(aes(x, y)) + 
  geom_point() + 
  xlab(NULL) +
  facet_wrap(~ name, nrow = 1)

p2 <-
  df %>% 
  filter(name %in% c("d", "e")) %>% 
  ggplot(aes(x, y)) + 
  geom_point() + 
  facet_wrap(~ name, nrow = 1) 

# Option 1 (no whitespace)
p1 / p2
```

![](https://i.imgur.com/sZCfIV3.png)

``` r
# Option 2 (with whitespace)
p1 + p2 & 
  plot_layout(design = "AAAAAAA
                        #######
                        #BBBBB#", 
              heights = c(3, 0.3, 3))
```

![](https://i.imgur.com/qrtD5Xx.png)
